### PR TITLE
Pass through database number

### PR DESCRIPTION
**Pull request by dschmidt**
In earlier versions options were just passed through, now database number needs to be passed through manually.

### CODE REVIEW

**Comment by lukemelia** [comment]
Do we need to fallback to a default value, or will an `undefined` database key be acceptable?

**Comment by dschmidt** [comment]
Well, currently no database number is passed anyway - I'm not too familiar with redis, I don't know if one can configure a default database to write to on server side or if it's always `0` for example. 

If one can't override it server side, it might make sense to add e.g. `0` as default, so it's more obvious it can be set.

**Comment by lukemelia** [comment]
Upon reflection on the various hosted redis services I've used, some of which give you only one database, I think we should include the option in the args to the underlying redis network client if and only if it is not `undefined`.

File: README.md [hunk 1]
```diff
@@ -75,6 +75,12 @@ The Redis port. If [url](#url) is defined, then this option is not needed.
 
 *Default:* `6379`
 
+### database
+
+The Redis database number.
+
+*Default:* `undefined`
```
Comment (lukemelia) [inline README.md]: Let's specify `If [url](#url) is defined, then this option is not needed.`

File: lib/redis.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ module.exports = CoreObject.extend({
     } else {
       var redisOptions = {
         host: options.host,
-        port: options.port
+        port: options.port,
+        database: options.database
```
Comment (lukemelia) [inline lib/redis.js]: Please only include the database key if `options.database !== undefined`

**Comment by lukemelia** [comment]
Can we add a test against the configure hook validating that this new option is accepted?

**Comment by lukemelia** [comment]
Looks good to me. Please squash to a single commit and @achambers can merge it in if he agrees.

**Comment by dschmidt** [comment]
Squashed.

**Comment by dschmidt** [comment]
If you're happy with the test, I'll squash again.

**Comment by achambers** [comment]
Thanks mate

**Pull request merged**